cmd/talosctl/cmd/talos/pcap.go

```diff
@@ -0,0 +1,196 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package talos
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/pcapgo"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/codes"
+
+	"github.com/talos-systems/talos/cmd/talosctl/pkg/talos/helpers"
+	"github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+)
+
+var pcapCmdFlags struct {
+	iface     string
+	promisc   bool
+	snaplen   int
+	output    string
+	bpfFilter string
+	duration  time.Duration
+}
+
+// pcapCmd represents the pcap command.
+var pcapCmd = &cobra.Command{
+	Use:     "pcap",
+	Aliases: []string{"tcpdump"},
+	Short:   "Capture the network packets from the node.",
+	Long: `The command launches packet capture on the node and streams back the packets as raw pcap file.
+
+Default behavior is to decode the packets with internal decoder to stdout:
+
+  talosctl pcap -i eth0
+
+Raw pcap file can be saved with --output flag:
+
+  talosctl pcap -i eth0 --output eth0.pcap
+
+Output can be piped to tcpdump:
+
+  talosctl pcap -i eth0 -o - | tcpdump -vvv -r -
+
+ BPF filter can be applied, but it has to compiled to BPF instructions first using tcpdump.
+ Correct link type should be specified for the tcpdump: EN10MB for Ethernet links and RAW
+ for e.g. Wireguard tunnels:
+
+   talosctl pcap -i eth0 --bpf-filter "$(tcpdump -dd -y EN10MB 'tcp and dst port 80')"
+
+   talosctl pcap -i kubespan --bpf-filter "$(tcpdump -dd -y RAW 'port 50000')"
+
+As packet capture is transmitted over the network, it is recommended to filter out the Talos API traffic,
+e.g. by excluding packets with the port 50000.
```

Comment:
probably good to have an example that filters traffic on port 50000

Reply:
also there might be some fun if the request is forwarded between apids, but still port 50000.

Reply:
I'm torn on that... We can only build a filter with tcpdump on client side. We need to know capture data type (-y). Client should have tcpdump installed.